packages/common/src/store/pages/chat/slice.ts

```diff
@@ -132,6 +128,41 @@ const slice = createSlice({
     },
     markChatAsRead: (_state, _action: PayloadAction<{ chatId: string }>) => {
       // triggers saga
+    },
+    sendMessage: (
+      _state,
+      _action: PayloadAction<{ chatId: string; message: string }>
+    ) => {
+      // triggers saga
+    },
+    addMessage: (
+      state,
+      action: PayloadAction<{ chatId: string; message: ChatMessage }>
+    ) => {
+      const { chatId, message } = action.payload
+      state.chatMessages[chatId].data.unshift(message)
+    },
+    upsertChat: (state, action: PayloadAction<{ chat: UserChat }>) => {
+      const { chat } = action.payload
+      const index = state.chatList.data.findIndex(
+        (c) => c.chat_id === chat.chat_id
+      )
+      if (index > -1) {
+        // remove existing
+        state.chatList.data.splice(index, 1)
+      }
+      // Assume new message and put at top of list
+      state.chatList.data.unshift(chat)
+    },
+    updateMessageReactions: (
+      state,
+      action: PayloadAction<{ chatId: string; message: ChatMessage }>
+    ) => {
+      const { chatId, message } = action.payload
+      const index = state.chatMessages[chatId].data.findIndex(
+        (m) => m.message_id === message.message_id
+      )
+      state.chatMessages[chatId].data[index].reactions = message.reactions
```

Comment:
guessing we're sure that index > -1 here?

Reply:
good call, will add some safeties